src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -364,6 +352,61 @@ private void MoveCaretToClosingPoint()
                 }
             }
 
+            private void ApplyBraceCompletionResult(BraceCompletionResult result)
+            {
+                using var edit = SubjectBuffer.CreateEdit();
+                foreach (var change in result.TextChanges)
+                {
+                    edit.Replace(change.Span.ToSpan(), change.NewText);
+                }
+
+                edit.ApplyAndLogExceptions();
+
+                Debug.Assert(SubjectBuffer.CurrentSnapshot[OpeningPoint.GetPosition(SubjectBuffer.CurrentSnapshot)] == OpeningBrace,
+                "The opening point does not match the opening brace character");
+                Debug.Assert(SubjectBuffer.CurrentSnapshot[ClosingPoint.GetPosition(SubjectBuffer.CurrentSnapshot) - 1] == ClosingBrace,
+                "The closing point does not match the closing brace character");
```

Comment:
consider making these NFWs.  we'll never hear about debug-asserts as no one runs in debug.